### PR TITLE
feat: add guild member

### PR DESCRIPTION
**Pull request by k1iin**
**Please describe the changes this PR makes and why it should be merged:**
I created an addMember method to the GuildsAPI, because I think someone needs this method

**Status and versioning classification:**
* This PR changes the library's interface (methods or parameters added)

### CODE REVIEW

File: packages/core/src/api/guild.ts [hunk 1]
```diff
@@ -164,6 +166,26 @@ export class GuildsAPI {
 	public async delete(guildId: Snowflake, { signal, reason }: Pick<RequestData, 'reason' | 'signal'> = {}) {
 		await this.rest.delete(Routes.guild(guildId), { reason, signal });
 	}
+	
+	/**
+	 * Add user to the guild
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:171]:
```suggestion
	 * Adds user to the guild
```

but still poor wording

File: packages/core/src/api/guild.ts [hunk 2]
```diff
@@ -164,6 +166,26 @@ export class GuildsAPI {
...
+         *
+         * @see {@link https://discord.com/developers/docs/resources/guild#add-guild-member}
+         * @param guildId - Id of the guild that will be added to the user's guilds
+	 * @param userId - Id of the user that will be added to guild
+         * @param body - The options for adding user to guild
+	 */
+	public async addMember(
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:178]: Doesnt it need a token?

Comment (k1iin) [reply packages/core/src/api/guild.ts:178]: yes, body parameter has `access_token` property

File: packages/core/src/api/guild.ts (hunk 1 above)
Comment (k1iin) [reply packages/core/src/api/guild.ts:171]: haha ik

File: packages/core/src/api/guild.ts (hunk 2 above)
Comment (jaw0r3k) [reply packages/core/src/api/guild.ts:178]: Okey then never mind

File: packages/core/src/api/guild.ts [hunk 3]
```diff
@@ -164,6 +166,26 @@ export class GuildsAPI {
 	public async delete(guildId: Snowflake, { signal, reason }: Pick<RequestData, 'reason' | 'signal'> = {}) {
 		await this.rest.delete(Routes.guild(guildId), { reason, signal });
 	}
+	
+	/**
+	 * Add user to the guild
+         *
+         * @see {@link https://discord.com/developers/docs/resources/guild#add-guild-member}
+         * @param guildId - Id of the guild that will be added to the user's guilds
+	 * @param userId - Id of the user that will be added to guild
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:175]:
```suggestion
The id of the user to add
```

File: packages/core/src/api/guild.ts [hunk 4]
```diff
@@ -164,6 +166,26 @@ export class GuildsAPI {
 	public async delete(guildId: Snowflake, { signal, reason }: Pick<RequestData, 'reason' | 'signal'> = {}) {
 		await this.rest.delete(Routes.guild(guildId), { reason, signal });
 	}
+	
+	/**
+	 * Add user to the guild
+         *
+         * @see {@link https://discord.com/developers/docs/resources/guild#add-guild-member}
+         * @param guildId - Id of the guild that will be added to the user's guilds
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:174]:
```sugggestion 
         * @param guildId - The id of the guild to add the user to
```
Or similar wording

File: packages/core/src/api/guild.ts [hunk 5]
```diff
@@ -164,6 +166,26 @@ export class GuildsAPI {
...
+	 * @param userId - Id of the user that will be added to guild
+         * @param body - The options for adding user to guild
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:176]:
```suggestion
         * @param body - The options for adding the user to a guild
```

File: packages/core/src/api/guild.ts [hunk 6]
```diff
@@ -168,12 +168,12 @@ export class GuildsAPI {
...
-	 * Add user to the guild
+	 * Adds user to the guild
          *
          * @see {@link https://discord.com/developers/docs/resources/guild#add-guild-member}
-         * @param guildId - Id of the guild that will be added to the user's guilds
-	 * @param userId - Id of the user that will be added to guild
-         * @param body - The options for adding user to guild
+         * @param guildId - The id of the guild to add the user to
+	 * @param userId - The id of the user to add
+         * @param body - The options for adding user to a guild
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:176]:
```suggestion
         * @param body - The options for adding the user to a guild
```

**Comment by Jiralite** [comment]
Should this be in `OAuth2API` since it's with respect to OAuth2?

**Comment by k1iin** [comment]
> Should this be in `OAuth2API` since it's with respect to OAuth2?

I was also thinking about that earlier, but I noticed that the `Routes.guildMember` endpoint is part of the Guild (in the dev portal docs). So, I added it to the GuildsAPI. Do I need to move this to the OAuth2API or not?

**Comment by jaw0r3k** [comment]
Also fix the formatting, as rest of the code uses tabs instead of spaces

File: packages/core/src/api/guild.ts [hunk 7]
```diff
@@ -165,6 +167,27 @@ export class GuildsAPI {
...
+	public async addMember(
+		guildId: Snowflake,
+		userId: Snowflake,
+		body: RESTPutAPIGuildMemberJSONBody,
+		{ signal }: Pick<RequestData, 'signal'> = {}
+	) {
+		return this.rest.put(Routes.guildMember(guildId, userId), {
+			body,
+			signal
+		}) as Promise<RESTPutAPIGuildMemberResult>
```
Comment (jaw0r3k) [inline packages/core/src/api/guild.ts:188]:
```suggestion
		}) as Promise<RESTPutAPIGuildMemberResult>;
```

formatting :>

**Comment by kyranet** [comment]
@k1iin can you run `pnpm run lint`, please? That'll resolve the issues reported by [CI](https://github.com/discordjs/discord.js/actions/runs/6449300077/job/17508309415?pr=9877#step:7:27).

**Review by vladfrangu — APPROVED** [review]

**Comment by Jiralite** [comment]
Please resolve conflicts.

**Review by Jiralite — CHANGES_REQUESTED** [review]
https://github.com/discordjs/discord.js/pull/9877#issuecomment-1752457192

**Comment by k1iin** [comment]
> @k1iin can you run `pnpm run lint`, please? That'll resolve the issues reported by [CI](https://github.com/discordjs/discord.js/actions/runs/6449300077/job/17508309415?pr=9877#step:7:27).

i got this warn when running `lint` 

```
 WARN  Unsupported engine: wanted: {"node":">=18.17.1"} (current: {"node":"v18.16.1","pnpm":"8.10.0"})

> @discordjs/discord.js@0.0.0 lint /home/runner/discordjs/discord.js
> turbo run lint --concurrency=4

 ELIFECYCLE  Command failed.
```

**Comment by k1iin** [comment]
> @k1iin can you run `pnpm run lint`, please? That'll resolve the issues reported by [CI](https://github.com/discordjs/discord.js/actions/runs/6449300077/job/17508309415?pr=9877#step:7:27).

i got this warn when running `lint`

```
> @discordjs/discord.js@0.0.0 lint /home/runner/discordjs/discord.js
> turbo run lint --concurrency=4

 ELIFECYCLE  Command failed.
```

**Comment by Jiralite** [comment]
That can't be the entire output... can you show it all?

**Review by kyranet — APPROVED** [review]

**Review by Jiralite — APPROVED** [review]

**Review by suneettipirneni — APPROVED** [review]

**Pull request merged**